api/python/provisioner/commands/upgrade/get_swupgrade_info.py

```diff
@@ -0,0 +1,142 @@
+
+from provisioner.salt import local_minion_id, cmd_run
+from provisioner.commands import CommandParserFillerMixin
+from provisioner.vendor import attr
+
+from . import SWUpgrade
+
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class GetSWUpgradeInfoArgs:
+    release: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "SW upgrade repository release version"
+            }
+        },
+        default=None
+    )
+
+
+class GetSWUpgradeInfo(CommandParserFillerMixin):
+    """
+    Base class that provides API for getting SW upgrade repository information.
+
+    """
+
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = GetSWUpgradeInfoArgs
+
+    @staticmethod
+    def _get_package_version(release: str) -> dict:
+        """
+        Function returns information about CORTX packages and their versions.
+
+        Parameters
+        ----------
+        release : str
+            SW upgrade repository version
+
+        Returns
+        -------
+        dict
+            return dictionary with CORTX packages and their versions
+
+        """
+        repo = f"sw_upgrade_{CORTX_ISO_DIR}_{release}"
+
+        cmd = (f"yum repo-pkgs {repo} list 2>/dev/null | "
+               f"grep '{repo}' | awk '{{print $1\" \"$2}}'")
+
+        res = cmd_run(cmd, targets=local_minion_id(),
+                      fun_kwargs=dict(python_shell=True))
+
+        packages = res[local_minion_id()].strip().split('\n')
+
+        if packages:
+            logger.debug(f"List of packages in repository '{repo}':"
+                         f" {packages}")
+        else:
+            logger.debug(f"There are no packages in repository '{repo}'")
+
+        res = dict()
+        # NOTE: Format is following
+        # ```
+        #  {
+        #      'cortx-motr': {
+        #             'version': '2.0.0-277',
+        #          },
+        #  }
+        # ```
+        #
+        # TODO: Along the with 'version', field we need to add
+        #  'constraint version' field to provide necessary information about
+        #  compatibility with old versions
+        for entry in packages:
+            pkg, ver = entry.split(" ")
+            res[pkg] = {SWUpgradeInfoFields.VERSION.value: ver}
+
+        return res
+
+    def run(self, release: str = None) -> dict:
+        """
+        Main function for Get SW Upgrade Repo command. Command returns
+        information about CORTX packages and their versions.
+
+        Parameters
+        ----------
+        release: str
+            SW upgrade repository release version
+
+        Returns
+        -------
+        dict:
+            return dictionary with CORTX packages and their versions
+
+        """
+        local_minion = local_minion_id()
+
+        if release is None:
+            # NOTE: take the latest release from SW upgrade repositories
+
+            # TODO: make get pillar API public and available for others to
```

Comment:
please consider [that](https://github.com/Seagate/cortx-prvsnr/blob/affa29980a3a5c5dc0e58d4b037de0d5b5f1595e/api/python/provisioner/commands/upgrade/sw_upgrade_node.py#L56) option, it should be sufficient. Also you may consider to fetch `release/upgrade/repos` key-path directly so you won't need to check  it against `None`.

Reply:
I updated code to use suggest approach. But anyway we need to have a convenient API to get pillar values without constructing long dict keys.

Reply:
Agree. Thanks